map/src/main/kotlin/edu/artic/map/helpers/CollectionExtensions.kt

```diff
@@ -0,0 +1,23 @@
+package edu.artic.map.helpers
+
+/**
+ * Simple inversion of [Iterable.contains].
+ */
+fun <T> Iterable<T>.doesNotContain(element: T): Boolean {
+    return !this.contains(element)
+}
+
+/**
+ * Copy of [java.util.Collection.removeIf]. Functionality is identical.
+ */
+fun <E> MutableCollection<E>.modifyThenRemoveIf(filter: (E) -> Boolean) : Boolean {
```

Comment:
theres no kotlin method to do this? theres a `filter` method which returns a list of filtered items matching the predicate.

Reply:
it returns a new list, but its better to deal with immutable lists in my opinion